### PR TITLE
E2E: Fix running any suite locally

### LINKED ISSUE #42178: Chore: fix running e2e tests locally
Awesome work on this 🎉  I think the experience locally with `yarn e2e` or `yarn e2e:dev` might not work any longer but we can fix that in a later PR.

### yarn e2e:dev
image

### yarn e2e
image

_Originally posted by @hugohaggmark in https://github.com/grafana/grafana/pull/41748#pullrequestreview-814439143_

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #42178

* Changes so that we can run tests from any suite (locally when no suite name is specified)  
* The new testFiles pattern removes the need for a nested folder for specs so unested the spec files 

Screenshot from 2021-11-24 12-23-22

### CODE REVIEW

File: e2e/run-suite [hunk 1]
```diff
@@ -14,20 +14,28 @@ CMD="start"
...
+    "dev")
+        echo "Dev mode"
+        CMD="open"
+        ;;
+    "")
+        ;;
+    *)
+        integrationFolder=../../e2e/"${args[0]}"
+        ;;
+esac
```
Comment (torkelo) [inline e2e/run-suite:37]: this is a maybe not ideal, but suite name "dev" means open cypress with all suite folders, and when no suite name specified we will run all specs

File: e2e/run-suite [hunk 2]
```diff
@@ -14,20 +14,28 @@ CMD="start"
...
+
 yarn $CMD --env BASE_URL=$URL,SLOWMO=$SLOWMO \
-  --config defaultCommandTimeout=30000,integrationFolder=../../e2e/"${args[0]}"/specs,screenshotsFolder=../../e2e"${args[0]}"/screenshots,videosFolder=../../e2e/"${args[0]}"/videos,fileServerFolder=./cypress,viewportWidth=1920,viewportHeight=1080,trashAssetsBeforeRuns=false,videoUploadOnPasses=false \
+  --config defaultCommandTimeout=30000,testFiles=*-suite/*spec.ts,integrationFolder=$integrationFolder,screenshotsFolder=../../e2e"${args[0]}"/screenshots,videosFolder=../../e2e/"${args[0]}"/videos,fileServerFolder=./cypress,viewportWidth=1920,viewportHeight=1080,trashAssetsBeforeRuns=false,videoUploadOnPasses=false \
```
Comment (dsotirakis) [inline e2e/run-suite:40]:
```suggestion
  --config defaultCommandTimeout=30000,testFiles=*-suite/*spec.ts,integrationFolder=$integrationFolder,screenshotsFolder=../../e2e/"${args[0]}"/screenshots,videosFolder=../../e2e/"${args[0]}"/videos,fileServerFolder=./cypress,viewportWidth=1920,viewportHeight=1080,trashAssetsBeforeRuns=false,videoUploadOnPasses=false \
```

**Review by hugohaggmark — APPROVED** [review]
LGTM 🎉

**Review by dsotirakis — APPROVED** [review]
Thanks for that @torkelo! 👏

**Comment by dsotirakis** [comment]
Adding backport label as well, I think it'll be great to have it in `v8.3.x` branch as well.

**Pull request merged**